untar.go

```diff
@@ -284,6 +294,11 @@ func UnTarIndex(ctx context.Context, dst string, index Index, s Store, n int, op
 	// Run untar in the main goroutine
 	err := UnTar(ctx, r, dst, opts)
 
+	if err != nil {
```

Comment:
With the `defer cancel()` in place above, all of this (to the end of the function) could be written as:
```go
	// Run untar in the main goroutine
	if err := UnTar(ctx, r, dst, opts); err != nil {
		return err
	}
	return g.Wait()
```

But thinking a bit more about this, what do you think about running the UnTar() inside a goroutine as well with g.Go() like the others? Then we wouldn't need the extra context, and there's no question about which error gets priority. All would be covered by `return g.Wait()` at the end. That should make things a little simpler.

Reply:
Thank you for such a nice idea; done.